app/models/task.rb

```diff
@@ -82,6 +82,6 @@ def destroy_incomplete_correction_submissions
   def set_due_on
     return if period_year.blank? || period_month.blank?
 
-    self.due_on ||= Task::ReportingPeriod.new(period_year, period_month).due_date
+    self.due_on ||= Task::SubmissionWindow.new(period_year, period_month).due_date
```

Comment:
If we're going to rename to `SubmissionWindow` (which is great as that's proper domain language), should we also rename `period_year` and `period_month` here to `submission_year` and `submission_month` respectively?

And if we do that, should we also do `generate.rake`?

Reply:
(or just `year`, `month`, as you've done in the specs)

Reply:
Maybe - although those names mirror the attributes in `Task` which is why I kept them here.